launchbar/org.eclipse.launchbar.core/src/org/eclipse/launchbar/core/internal/LaunchBarManager.java

```diff
@@ -140,11 +140,13 @@ void init() throws CoreException {
 				ILaunchDescriptor last = null;
 				for (String id : split) {
 					Pair<String, String> key = toId(id);
-					ILaunchDescriptor desc = descriptors.get(key);
-					if (desc != null) {
-						descriptors.remove(key);
```

Comment:
I think the remove/put call is redundant?

Reply:
I believe this is to keep the order in the `LinkedHashMap` as expected, since there is other code that depends on the order of adding elements to the map. It might be possible to replace the 2 calls with a single call to another API, though I'm not 100% the order will be the right one. It will have to be double-checked.

Reply:
Good point, I had overlooked that use of remove/put. I wouldn't spend more time on it than that.